AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -365,6 +388,30 @@ - (void)setImage:(UIImage *)image forState:(ASControlState)state
   [self updateImage];
 }
 
+- (UIImage *)backgroundImageForState:(ASControlState)state
+{
+    ASDN::MutexLocker l(_propertyLock);
+    switch (state) {
+      case ASControlStateNormal:
+        return _normalBackgroundImage;
+            
+      case ASControlStateHighlighted:
+        return _highlightedBackgroundImage;
+            
+      case ASControlStateSelected:
+        return _selectedBackgroundImage;
+            
+      case ASControlStateSelected | ASControlStateHighlighted:
+        return _selectedHighlightedBackgroundImage;
+      
+      case ASControlStateDisabled:
+        return _disabledBackgroundImage;
+            
+      default:
+        return _normalBackgroundImage;
+    }
+}
+
```

Comment:
@tomizimobile Seems like the spacing for this has reverted to 4 spaces. The standard for ASDK is 2 spaces. You can change this in your project settings. I fell for this a lot initially.